### PR TITLE
chore: move CI jobs to GH actions

### PR DESCRIPTION
**Pull request by mndeveci**
Moves CI jobs to GH Actions.

- Removed individual appveyor configurations that is used by CI jobs
- Updated `build.yml` file to have integration tests + functional & smoke tests
- Updated individual test cases to run on Windows as well (they were managed in different files before. See [appveyor-windows-build-dotnet.yml](https://github.com/aws/aws-sam-cli/compare/develop...mndeveci:aws-sam-cli:move_gh_actions_for_ci_jobs?expand=1#diff-db702ea1aa763c9113b929bb642263b9e683b2bfdcceb967002a36e19b5470a3) for instance)

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,22 +9,24 @@ on:
 jobs:
   run-workflow:
     name: PR Workflow
-    if: always()
+    if: github.repository_owner == 'aws'
     runs-on: ubuntu-latest
     needs:
```
Comment (mildaniel) [inline .github/workflows/build.yml:14]: Should `smoke-and-functional-tests` be added under the `needs` section too?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -62,21 +64,53 @@ jobs:
       matrix:
         os:
           - ubuntu-latest
-          # TODO(hawflau): investigate and fix flakiness on Windows
-          # - windows-latest
+          - windows-latest
```
Comment (mildaniel) [inline .github/workflows/build.yml:67]: Did the Windows flakiness get resolved?

File: tests/integration/buildcmd/test_build_cmd_rust.py [hunk 3]
```diff
@@ -26,9 +26,6 @@
 
 LOG = logging.getLogger(__name__)
 
-# SAR tests require credentials. This is to skip running the test where credentials are not available.
-SKIP_SAR_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
```
Comment (mildaniel) [inline tests/integration/buildcmd/test_build_cmd_rust.py:30]: Was this just not used anywhere?

**Review by mildaniel — COMMENTED** [review]
This is exciting, thanks for doing this!

**Comment by sriram-mv** [comment]
Super Excited for this!

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -62,21 +65,53 @@ jobs:
...
         python:
           - "3.7"
           - "3.8"
           - "3.9"
-        rust:
-          - stable
+        # folders that is commented below requires credentials, no need to spare time to run them
+        tests_folder:
+          - "buildcmd"
```
[the remark this answers is not in the document]
Comment (mndeveci) [reply .github/workflows/build.yml:75]: We have skip configuration for the tests (or part of the test) where we require docker. I saw some of them were missing so updated them with my latest commit.

File: .github/workflows/build.yml (hunk 2 above)
Comment (mndeveci) [reply .github/workflows/build.yml:68]: I've confirmed with @hawflau the failures are coming from the test runs (not during tooling setup). See example: https://github.com/aws/aws-sam-cli/actions/runs/4207102433/jobs/7301465033

And rust tests are configured to skip windows CI environment at the moment.

**Review by mildaniel — APPROVED** [review]

**Comment by sriram-mv** [comment]
There still seem to be windows flakiness. I'm approving, but those need to be fixed before merging.

**Review by sriram-mv — APPROVED** [review]

**Comment by mndeveci** [comment]
> There still seem to be windows flakiness. I'm approving, but those need to be fixed before merging.

Yeah I am still trying to figure out places where we make a docker api call. I am cleaning them one by one.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -62,21 +65,53 @@ jobs:
...
-          # - windows-latest
+          - windows-latest
         python:
           - "3.7"
           - "3.8"
           - "3.9"
-        rust:
-          - stable
+        # folders that is commented below requires credentials, no need to spare time to run them
+        tests_folder:
```
Comment (hawflau) [inline .github/workflows/build.yml:74]: (non-blocking) question - do we have any plan to load balance test load under each test folder or across all test cases? My worry is some folders contains lot more test cases than others. So overall running all tests under that particular test folder will take much more time than others combined

Comment (mndeveci) [reply .github/workflows/build.yml:74]: That is a valid point. I think we can give a chance to `pytest-split` (https://jerry-git.github.io/pytest-split/) which creates test groups from their time to completion.

**Review by mildaniel — APPROVED** [review]

File: .test_durations [hunk 6]
```diff
@@ -0,0 +1,1816 @@
+{
```
Comment (sriram-mv) [inline .test_durations:1]: is this file required?

**Review by sriram-mv — APPROVED** [review]

File: .test_durations (hunk 6 above)
Comment (mndeveci) [reply .test_durations]: It was coming from `pytest-split` library, but I removed it now until we will have a clear way to use it (since we need to maintain up to date information about how much time each test takes).
